libs/core/tests/unit_tests/utils/test_function_calling.py

```diff
@@ -296,60 +296,59 @@ def test_convert_to_openai_function(
     assert actual == runnable_expected
 
 
+@pytest.mark.xfail(reason="Direct pydantic v2 models not yet supported")
+def test_convert_to_openai_function_nested_v2() -> None:
+    class NestedV2(BaseModelV2Maybe):
+        nested_v2_arg1: int = FieldV2Maybe(..., description="foo")
+        nested_v2_arg2: Literal["bar", "baz"] = FieldV2Maybe(
+            ..., description="one of 'bar', 'baz'"
+        )
+
+    def my_function(arg1: NestedV2) -> None:
+        """dummy function"""
+        pass
+
+    convert_to_openai_function(my_function)
+
+
 def test_convert_to_openai_function_nested() -> None:
     class Nested(BaseModel):
         nested_arg1: int = Field(..., description="foo")
         nested_arg2: Literal["bar", "baz"] = Field(
             ..., description="one of 'bar', 'baz'"
         )
 
-    class NestedV2(BaseModelV2Maybe):
-        nested_v2_arg1: int = FieldV2Maybe(..., description="foo")
-        nested_v2_arg2: Literal["bar", "baz"] = FieldV2Maybe(
-            ..., description="one of 'bar', 'baz'"
-        )
-
-    def my_function(arg1: Nested, arg2: NestedV2) -> None:
+    def my_function(arg1: Nested) -> None:
         """dummy function"""
         pass
 
-        expected = {
```

Comment:
this was indented so it was part of my_function and not actually tested, which is why it wasn't failing